### PR TITLE
[CI] Restore to params.MIGraphXBranch

### PR DESCRIPTION
**Pull request by manupak**
This commit restores the CI to verify
against tip of MIGraphX.

### CODE REVIEW

**Comment by krzysz00** [comment]
Why? We currently can't build against tip

**Comment by manupak** [comment]
This is not to build against tip but to build against params.MIGraphXBranch -- whichever that is now (which I think what you changed -- without this PR your change has no effect)
you can see there are interceptors that have sneaked in.

**Comment by manupak** [comment]
MIGraphX stage is green : http://rocmhead:8080/blue/organizations/jenkins/MLIR%2Fmlir/detail/PR-1146/2/pipeline/922

**Comment by krzysz00** [comment]
... having looked, I found a bug in the MIGraphX stage.

The CMAKE_PREFIX_PATH needs to list our workspace before the `/MIGraphXDeps` directory, since that directory could contain a MLIR copy.

**Comment by manupak** [comment]
Agreed -- and a good catch. 
Since MIGraphX dont list us as a dependency (yet), I'd argue its unlikely for MLIR to be in there for timebeing.

Having checked MIGraphX stage passing on this PR, I'd like to get this in to unblock nightlies.

**Comment by krzysz00** [comment]
Given that https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/1146 was needed after upstream merge, it's surprising that MIGraphX CI passes.

But I'll land this anyway in exchange for you joining me in getting this all untangled.

File: mlir/utils/jenkins/Jenkinsfile
Comment (krzysz00) [inline mlir/utils/jenkins/Jenkinsfile:56]: ... I'll note that this doesn't actually change the branch to `develop`, but instead makes the branch my upstream branch PR.

**Review by krzysz00 — APPROVED** [review]
Having looked at the PR, this change makes my PR from Friday actually do what it says it does, hence everything working.

Approved.

**Pull request merged**
